### PR TITLE
Adding and checking for supported encodings in util function to avoid exception

### PR DESCRIPTION
**Pull request by issacnitin**
PR to fix Issue https://github.com/microsoft/typed-rest-client/issues/203

### CODE REVIEW

**Comment by damccorm** [comment]
@isaacnitin - where is this list of supported encodings coming from?

**Comment by issacnitin** [comment]
@damccorm These are the officially supported encodings of nodejs

File: lib/Util.ts [hunk 1]
```diff
@@ -113,9 +113,9 @@ export function obtainContentCharset (response: IHttpClientResponse) : string {
   // |__ matches would be ['charset=utf-8', 'utf-8', index: 18, input: 'application/json; charset=utf-8']
   // |_____ matches[1] would have the charset :tada: , in our example it's utf-8
   // However, if the matches Array was empty or no charset found, 'utf-8' would be returned by default.
-
+  const supportedEncodings = ['ascii', 'utf8', 'utf16le', 'ucs2', 'base64', 'binary', 'hex']
```
Comment (stephenmichaelf) [inline lib/Util.ts]: Rename `nodeSupportedEncodings`?

**Comment by stephenmichaelf** [comment]
We should add tests for this.

**Comment by issacnitin** [comment]
@stephenmichaelf I've added Unit tests and renamed the variable.

**Comment by damccorm** [comment]
This has been released - https://www.npmjs.com/package/typed-rest-client

**Pull request merged**